examples/satellites/check_stability.py

```diff
@@ -0,0 +1,188 @@
+              'bool', in_top_ten('DSP 20 (USA 149)')),
+             ("SDS III-6 in top 10 unlikely periods",
+              'bool', in_top_ten('SDS III-6')),
+             ("SDS III-7 in top 10 unlikely periods",
+              'bool', in_top_ten('SDS III-7')),
+             ("Orion 6 in top 10 unlikely periods",
+              'bool', in_top_ten('Advanced Orion 6')),
+             ("DSP 20 in top 5 unlikely periods",
+              'bool', in_top_five('DSP 20 (USA 149)')),
+             ("SDS III-6 in top 5 unlikely periods",
+              'bool', in_top_five('SDS III-6')),
+             ("SDS III-7 in top 5 unlikely periods",
+              'bool', in_top_five('SDS III-7')),
+             ("Orion 6 in top 5 unlikely periods",
+              'bool', in_top_five('Advanced Orion 6')),
+         ]
+
+def orbit_type_imputation_probes(bdb):
+    bdb.execute('''
+        CREATE TEMP TABLE inferred_orbit AS
+        INFER EXPLICIT
+        anticipated_lifetime, perigee_km, period_minutes, class_of_orbit,
+        PREDICT type_of_orbit
+            AS inferred_orbit_type
+            CONFIDENCE inferred_orbit_type_conf
+        FROM satellites_cc
+        WHERE type_of_orbit IS NULL;
+    ''')
+    query = '''
+        SELECT i.class_of_orbit, i.inferred_orbit_type,
+            COUNT(*) AS count, AVG(i.inferred_orbit_type_conf) AS mean,
+            AVG((i.inferred_orbit_type_conf - sub.answer) * (i.inferred_orbit_type_conf - sub.answer)) as variance
+        FROM inferred_orbit AS i,
+            (SELECT i.class_of_orbit, i.inferred_orbit_type,
+                AVG(i.inferred_orbit_type_conf) AS answer
+            FROM inferred_orbit as i
+            GROUP BY i.class_of_orbit, i.inferred_orbit_type) AS sub
+        WHERE i.class_of_orbit == sub.class_of_orbit
+            AND i.inferred_orbit_type == sub.inferred_orbit_type
+        GROUP BY i.class_of_orbit, i.inferred_orbit_type
+        ORDER BY i.class_of_orbit, i.inferred_orbit_type
+    '''
+    answers = dict(((cl, tp), (ct, avg, var))
+                   for (cl, tp, ct, avg, var)
+                   in bdb.execute(query).fetchall())
+    def probe_gen():
+        for cl in ["Elliptical", "LEO"]:
+            for tp in ["Deep Highly Eccentric",
+                       "Intermediate",
+                       "Molniya",
+                       "Sun-Synchronous",
+                       "Cislunar",
+                       "N/A"]:
+                (ct, avg, var) = answers.get((cl, tp), (0,0,0))
+                yield ("%s %s ct imputed instances" % (cl, tp),
+                       'num', float(ct))
+                yield ("%s %s mean inference confidence" % (cl, tp),
+                       'num', avg)
+                yield ("%s %s inference confidence stddev" % (cl, tp),
+                       'num', math.sqrt(var))
+    return list(probe_gen())
+
+######################################################################
+## Driver                                                           ##
```

Comment:
Conceivably, but I am loath to standardize on Pickle as a disk format.  And the analyze-and-do-not-save interface still needs to be exposed, so the extra doesn't seem all that helpful.  Especially since I can't make plotting symmetric without either hardcoding or parameterizing on the file naming convention (from which analysis counts are to be extracted).  Suggestions?